pkg/types/validation/installconfig.go

```diff
@@ -131,6 +131,16 @@ func ValidateInstallConfig(c *types.InstallConfig) field.ErrorList {
 	if _, ok := validPublishingStrategies[c.Publish]; !ok {
 		allErrs = append(allErrs, field.NotSupported(field.NewPath("publish"), c.Publish, validPublishingStrategyValues))
 	}
+	allErrs = append(allErrs, validateImageDigestSources(c.ImageDigestSources, field.NewPath("imageDigestSources"))...)
+	if _, ok := validPublishingStrategies[c.Publish]; !ok {
+		allErrs = append(allErrs, field.NotSupported(field.NewPath("publish"), c.Publish, validPublishingStrategyValues))
+	}
+	if len(c.ImageContentSources) != 0 {
+		logrus.Warningln("ImageContentSource is deprecated, please use ImageDigestSource")
```

Comment:
I wouldn't print this warning when you're already going to error out because they supplied both fields.

If they supply both, just print the error.

Reply:
fixed